src/topology/sheaves/sheaf_condition/unique_gluing.lean

```diff
@@ -0,0 +1,198 @@
+agree, for all `i` and `j`
+-/
+def is_compatible (sf : Π i : ι, F.obj (op (U i))) : Prop :=
+  ∀ i j : ι, F.map (inf_le_left (U i) (U j)).op (sf i) = F.map (inf_le_right (U i) (U j)).op (sf j)
+
+/--
+For presheaves of types, terms of `pi_opens F U` are just families of sections
+-/
+def pi_opens_iso_sections_family : pi_opens F U ≅ Π i : ι, F.obj (op (U i)) :=
+  limits.is_limit.cone_point_unique_up_to_iso
+    (limit.is_limit (discrete.functor (λ i : ι, F.obj (op (U i)))))
+    ((types.product_limit_cone (λ i : ι, F.obj (op (U i)))).is_limit)
+
+/--
+Under the isomorphism `pi_opens_iso_sections_family`, compatibility of sections is the same
+as being equalized by the arrows `left_res` and `right_res` of the equalizer diagram.
+-/
+lemma compatible_iff_left_res_eq_right_res (sf : pi_opens F U) :
+  is_compatible F U ((pi_opens_iso_sections_family F U).hom sf)
+    ↔ left_res F U sf = right_res F U sf :=
+begin
+  split ; intros h,
+  { ext ⟨i,j⟩,
+    rw [left_res, types.limit.lift_π_apply, fan.mk_π_app,
+        right_res, types.limit.lift_π_apply, fan.mk_π_app],
+    exact h i j, },
+  { intros i j,
+    convert congr_arg (limits.pi.π (λ p : ι × ι, F.obj (op (U p.1 ⊓ U p.2))) (i,j)) h,
+    {rw [left_res, types.pi_lift_π_apply], refl},
+    {rw [right_res, types.pi_lift_π_apply], refl},
+  }
+end
+
+/--
+A section `s` is a gluing for a family of sections `sf` if it restricts to `sf i` on `U i`,
+for all `i`
+-/
+def is_gluing (sf : Π i : ι, F.obj (op (U i))) (s : F.obj (op (supr U))) : Prop :=
+  ∀ i : ι, F.map (opens.le_supr U i).op s = sf i
+
+/--
+Under the isomorphism `pi_opens_iso_sections_family`, being a gluing of a family of
+sections `sf` is the same as lying in the preimage of `res` (the leftmost arrow of the
+equalizer diagram).
+-/
+@[simp]
+lemma is_gluing_iff_eq_res (sf : pi_opens F U) (s : F.obj (op (supr U))):
+  is_gluing F U ((pi_opens_iso_sections_family F U).hom sf) s ↔ res F U s = sf :=
+begin
+  split ; intros h,
+  { ext i,
+    rw [res, types.limit.lift_π_apply, fan.mk_π_app],
+    exact h i, },
+  { intro i,
+    convert congr_arg (limits.pi.π (λ i : ι, F.obj (op (U i))) i) h,
+    rw [res, types.pi_lift_π_apply] },
+end
+
+/--
+The subtype of all gluings for a given family of sections
+-/
+@[nolint has_inhabited_instance]
+def gluing (sf : Π i : ι, F.obj (op (U i))) : Type u :=
+  {s : F.obj (op (supr U)) // is_gluing F U sf s}
+
+/--
+The sheaf condition of type-valued presheaves in terms of unique gluings. A presheaf
+`F : presheaf (Type u) X` satisfies this sheaf condition if and only if, for every
+compatible family of sections `sf : Π i : ι, F.obj (op (U i))`, there exists a unique
+gluing `s : F.obj (op (supr U))`.
+
+We prove this to be equivalent to the usual one below in
+`sheaf_condition_equiv_sheaf_condition_unique_gluing`
+-/
+@[derive subsingleton, nolint has_inhabited_instance]
+def sheaf_condition_unique_gluing : Type (u+1) :=
```

Comment:
Would it make sense to call this `unique_gluing_condition`? It's a bit shorter, and the docstring explains that it's just another sheaf condition.

Reply:
A shorter name would be fine by me, I was just following the names of the existing sheaf conditions, e.g. `sheaf_condition_pairwise_intersections` and `sheaf_condition_opens_le_cover`

Reply:
Aah, that makes sense. Let's leave it like this.